app/feeswapper/management/commands/feeSwapper.py

```diff
@@ -0,0 +1,185 @@
+                print(r.json())
+
+                # Check if API is functioning or not and return null token list if so
+                if not r.ok:
+                        return {}
+                # Check if API call returned any ERC-20 tokens associated with the FEE_ADDRESS.  Return null token list if not
+                if r.json()['status'] != '1':
+                        return {}
+                for transaction in r.json()['result']:
+                        address = transaction['contractAddress']                      
+                        exchangeAddress = self.factoryContract.functions.getExchange(self.web3.toChecksumAddress(address)).call()
+                        if exchangeAddress != '0x0000000000000000000000000000000000000000':
+                                # Only add token to list to ETH if Uniswap exchange exists
+                                tokenList[address]={'tokenName':transaction['name'],'tokenSymbol':transaction['symbol'],'exchangeAddress':exchangeAddress,'balance':transaction['balance']}
+                                print('Token Name: ' + tokenList[address]['tokenName']+ ' and Token Symbol is: ' + tokenList[address]['tokenSymbol'])
+                                print('Token address is: ' + address)
+                                print('Uniswap Exchange address is: '+ exchangeAddress)
+                                print('Token balance is: ' + tokenList[address]['balance'])
+                return tokenList
+
+        def sell_token(self, exchangeAddress, tokenSymbol):
+                """Swap total balance of ERC-20 token associated with Uniswap exchange address to ETH
+                
+                Args:
+                        exchangeAddress (str): The address of the Uniswap exchange contract assocciated with the ERC-20 token to convert to ETH
+                        tokenSymbol (str): The symbol for the ERC-20 token to be converted to ETH. 
+                
+                Note: This currently only works with the BAT Uniswap exchange on Rinkeby Testnet
+                """
+                if (self.network == 'rinkeby'):
+                        chain = 4
+                else:
+                        chain = 1
+                # Follows Uniswap doc guidance on calculating conversion
+                exchangeContract = self.web3.eth.contract(address = exchangeAddress, abi = exchangeABI)
+                tokenAddress = exchangeContract.functions.tokenAddress().call()
+                tokenContract = self.web3.eth.contract(address = tokenAddress, abi = tokenABI)
+                walletBalance = tokenContract.functions.balanceOf(settings.FEE_ADDRESS).call()
+                if walletBalance == 0:
+                        return
+                outputReserve = self.web3.eth.getBalance(exchangeAddress)
+                inputReserve = tokenContract.functions.balanceOf(exchangeAddress).call()
+                numerator = walletBalance * outputReserve * 997
```

Comment:
Theres some magic numbers here

Reply:
Don't disagree but I'm following the [uniswap documentation](https://docs.uniswap.io/frontend-integration/swap#amount-bought-sell-order) exactly on this one.  Do you want these as additional fields in settings.py?  I could do a general Uniswap section.

Reply:
I added two additional fields to settings, UNISWAP_LIQUIDITY_FEE which is the 0.3% fee that Uniswap charges on trades and is built into the trade calculation formula.  It's hardcoded in the Uniswap conrtacts but could be updated if we ever wanted to call some future version of the Uniswap contract if they introduce a new version with a dynamic fee structure.
The second is UNISWAP_TRADE_LIMIT.  This is used to determine how long the sell order will be valid on the blockchain and defaults to 5 minutes (per the Uniswap guidelines).  Are there any other hardcoded numbers you're wondering about?  I tried to only leave in the powers of 10 that shift decimal points around.